test/test_formatters.rb

```diff
@@ -63,6 +63,42 @@ def test_vim
     ))
   end
 
+  def test_vim_with_line_numbers
+    vim = formatter_for(:format => 'vim', :fields => %w(n).to_set)
+    assert_equal %{C\t./script.rb\t/^class C < D$/;"\tc\tline:1}, vim.format(build_tag(
+      :kind => 'class', :name => 'C',
+      :pattern => "class C < D"
+    ))
+    assert_equal %{C\t./script.rb\t/^class C < D$/;"\tc\tline:42}, vim.format(build_tag(
+      :kind => 'class', :name => 'C',
+      :pattern => "class C < D",
+      :line => 42
+    ))
+    assert_equal %{C\t./script.rb\t/^class C < D$/;"\tc\tline:105499}, vim.format(build_tag(
+      :kind => 'class', :name => 'C',
+      :pattern => "class C < D",
+      :line => 105499
+    ))
+  end
+
+  def test_vim_with_excmd_number
+    vim = formatter_for(:format => 'vim', :excmd => "number")
+    assert_equal %{C\t./script.rb\t/^class C < D$/;"\t1;"\tc}, vim.format(build_tag(
```

Comment:
The `--excmd=number` option enables a mode where pattern is no longer output (e.g. there is no `/^class C < D$/`). In its place, the line number is output.